test/trace/integrations/rails_integration_test.rb

```diff
@@ -0,0 +1,129 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "fileutils"
+require "open3"
+require "timeout"
+require "test_helper"
+require "opencensus/trace/integrations/rails"
+require "faraday"
+
+describe "Rails integration" do
+  module RailsTestHelper
+    APP_NAME = "railstest"
+    BASE_DIR = File.absolute_path File.dirname File.dirname File.dirname __dir__
+    TMP_DIR = File.join BASE_DIR, "tmp"
+    APP_DIR = File.join TMP_DIR, APP_NAME
+    RAILS_OPTIONS = %w(
+        --skip-yarn
+        --skip-action-mailer
+        --skip-active-record
+        --skip-action-cable
+        --skip-puma
+        --skip-sprockets
+        --skip-spring
+        --skip-listen
+        --skip-coffee
+        --skip-javascript
+        --skip-turbolinks
+        --skip-test
+        --skip-system-test
+        --skip-bundle
+      ).join ' '
+
+    class << self
+      def create_rails_app
+        puts "**** Creating test Rails app..."
+        FileUtils.mkdir_p TMP_DIR
+        FileUtils.rm_rf APP_DIR
+        Dir.chdir TMP_DIR do
+          system "bundle exec rails new #{APP_NAME} #{RAILS_OPTIONS}"
+        end
+        Dir.chdir APP_DIR do
+          File.open "Gemfile", "a" do |f|
+            f.puts "gem 'opencensus', path: '#{BASE_DIR}'"
+          end
+          insert_after "config/application.rb", /Bundler\.require/,
+                       ["require 'opencensus/trace/integrations/rails'",
+                        "OpenCensus::Trace.configure.exporter = ",
+                        "  OpenCensus::Trace::Exporters::Logger.new(",
+                        "    ::Logger.new(STDERR, ::Logger::INFO))"]
+          insert_after "config/routes.rb", /routes\.draw/, "  root 'home#index'"
+          File.open "app/controllers/home_controller.rb", "w" do |f|
+            f.puts "class HomeController < ApplicationController"
+            f.puts "  def index; render plain: 'OK'; end"
+            f.puts "end"
+          end
+          Bundler.with_original_env do
+            system "bundle lock"
+            system "bundle install --deployment --gemfile Gemfile"
+          end
+        end
+        puts "**** Finished creating test Rails app"
+      end
+
+      def insert_after path, regex, new_lines
+        data = File.read path
+        File.open path, "w" do |file|
+          data.each_line do |line|
+            file.puts line
+            if regex && regex =~ line
+              Array(new_lines).each { |new_line| file.puts new_line }
+              regex = nil
+            end
+          end
+        end
+        raise "Unable to find #{regex.inspect} in #{path.inspect}" if regex
+      end
+
+      def run_rails_app timeout: 5
+        Dir.chdir APP_DIR do
+          Bundler.with_original_env do
+            Open3.popen2e "bundle exec rails s -p 3000" do |_in, out, thr|
+              begin
+                Timeout.timeout 5 do
```

Comment:
did you mean to use the `timeout` argument here?

Reply:
It's a positional argument, not a keyword argument. https://ruby-doc.org/stdlib-2.4.2/libdoc/timeout/rdoc/Timeout.html#method-c-timeout

Reply:
I meant from this method's arguments: `def run_rails_app timeout: 5`

Reply:
Oh, whoops. Fixed.